core/server/api/v2/redirects.js

```diff
@@ -0,0 +1,98 @@
+const fs = require('fs-extra');
+const path = require('path');
+const Promise = require('bluebird');
+const moment = require('moment-timezone');
+const config = require('../../config');
+const common = require('../../lib/common');
+const validation = require('../../data/validation');
+const web = require('../../web');
+
+const _private = {};
+
+_private.readRedirectsFile = (customRedirectsPath) => {
+    const redirectsPath = customRedirectsPath || path.join(config.getContentPath('data'), 'redirects.json');
+
+    return fs.readFile(redirectsPath, 'utf-8')
+        .then((content) => {
+            try {
+                content = JSON.parse(content);
+            } catch (err) {
+                throw new common.errors.BadRequestError({
+                    message: common.i18n.t('errors.general.jsonParse', {context: err.message})
+                });
+            }
+
+            return content;
+        })
+        .catch((err) => {
+            if (err.code === 'ENOENT') {
+                return Promise.resolve([]);
+            }
+
+            if (common.errors.utils.isIgnitionError(err)) {
+                throw err;
+            }
+
+            throw new common.errors.NotFoundError({
+                err: err
+            });
+        });
+};
+
+module.exports = {
+    docName: 'redirects',
+
+    download: {
+        headers: {
+            disposition: {
+                type: 'file',
+                value: 'redirects.json'
```

Comment:
Is the value here correct? 🤔 e.g. take a look at settings ctrl.

Reply:
Yeah, this is an acceptable format as well, the [test](https://github.com/TryGhost/Ghost/pull/10053/files#diff-206ddd0fb7e274adca57f6cea2126ff7R44) proves it works. 

The header in settings controller is for 'yaml' format, which is handled a bit [differently](https://github.com/TryGhost/Ghost/blob/master/core/server/api/shared/headers.js#L36) to the one for [file](https://github.com/TryGhost/Ghost/blob/master/core/server/api/shared/headers.js#L67) type.

Reply:
👍 shouldn't we normalize it?

Reply:
Agree, we could discuss a way to have all the file content disposition options to be me sorted. 

I think to keep the code on controller level to the minimum we could extract all the unnecessary: `Attachment; filename=""` parts. So for other controllers the end result could look like:
```javascript
           //  subscribers controller
            disposition: {
                type: 'csv',
                value() {
                    const datetime = (new Date()).toJSON().substring(0, 10);
                    return `"subscribers.${datetime}.csv"`;
                }
            }
```
and in the headers it would be:
```
'Content-Disposition': `Attachment; filename="${filename}"`
```
Don't know if we should go as far as handling the file extension in the headers module? :thinking: What do you think?

Reply:
> I think to keep the code on controller level to the minimum we could extract all the unnecessary: Attachment; filename="" parts. 

Yeah that makes sense i think 👍 I was wondering about this too a while back.

Reply:
Will create a tiny issue for this, so this PR doesn't become bloated :)

Reply:
This small refactor will be handled here - https://github.com/TryGhost/Ghost/issues/10331